### ISSUE TITLE
`Global name "x" :|[] not in scope` error for simple identity function.

### ISSUE THREAD

**Issue by systemf-frank** [anoma/juvix#809]
When defining a simple, non polymorphic identity function, type checking fails:
```ocaml
mod Identity where

open Prelude
open Prelude.Michelson

sig intid : int -> int
let intid x = x

sig main : int -> int
let main = intid
```
Apparently, `x` is recognised as a global, which is incorrect:
```
Elim' (Free' (Global ("Identity" :| ["intid"])) ()) ()
"TypecheckerError Global name "x" :| [] not in scope"
```
Note that there is not global called `x` at all, so it is not an issue of variable shadowing.

Adding an unused polymorphic argument:
```ocaml
sig intid : (x : ty) -> int -> int
let intid _ x = x

sig main : int -> int
let main = intid int
```
gives us the same result:
```
Elim' (App' (Free' (Global ("Identity" :| ["intid"])) ()) (Elim' (Free' (Global ("Prelude" :| ["Michelson","int"])) ()) ()) ()) ()
"TypecheckerError Global name "x" :| [] not in scope"
```

Making the function fully polymorphic, gives us (obviously) the error described in #799.

**Comment by systemf-frank** [comment]
Interestingly, rewriting it with an alias for `intid`:
```ocaml
sig intid : int -> int
let intid x = x

sig intid2 : int -> int
let intid2 = intid

sig main : int -> int
let main = intid2
```
Gives us a new error:
```
"InvalidInputType "Ty (TLambda (Ty TInt :) (Ty TInt :)) :""
```
But I don't think this is related to this.

For clarification, this is without the initial fix in #805. I am not sure if a proper fix for that issue could fix this particular error as well. I am pretty sure the 'not in scope' in the first post is unrelated to #805.

**Comment by systemf-frank** [comment]
The initial program:
```ocaml
sig intid : int -> int
let intid x = x

sig main : int -> int
let main = intid
```
Works fine now. The one with the added polymorphism fail for a reason unrelated to this issue. The code using the `intid2` alias works as well.

This issue can be closed now.
